hudi-common/src/main/java/org/apache/hudi/common/model/OverwriteWithLatestAvroPayload.java

```diff
@@ -59,24 +59,28 @@ public OverwriteWithLatestAvroPayload preCombine(OverwriteWithLatestAvroPayload
 
   @Override
   public Option<IndexedRecord> combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema) throws IOException {
+    return getInsertValue(schema);
+  }
 
-    Option<IndexedRecord> recordOption = getInsertValue(schema);
-    if (!recordOption.isPresent()) {
+  @Override
+  public Option<IndexedRecord> getInsertValue(Schema schema) throws IOException {
```

Comment:
Trying to understand what we are really fixing here... 
If there were an insert and delete in the same batch, precombining favors delete, then we actually skip the record and not do the insert?

Reply:
have updated the description.